api/fossa/uploadv1.go

```diff
@@ -0,0 +1,93 @@
+package fossa
+
+import (
+	"encoding/json"
+	"net/url"
+
+	"github.com/pkg/errors"
+
+	"github.com/apex/log"
+	"github.com/fossas/fossa-cli/cmd/fossa/version"
+)
+
+type V1UploadBody struct {
+	Vcs struct {
+		Type      string
+		Reference string
+	}
+	Project struct {
+		Title string
+		Id    string
+		Url   string
+		Jira  string
+		Team  string
+	}
+	Revision struct {
+		Id   string
+		Link string
+	}
+	Meta struct {
+		v string
+	}
+	Analysis []SourceUnit
+}
+
+// Upload uploads a project's analysis.
+func UploadV1(uploadBody V1UploadBody) (Locator, error) {
+	log.Info("Uploading build using API v1")
+	// Check preconditions.
+	if uploadBody.Revision.Id == "" {
+		return Locator{}, errors.New("Could not infer revision name from `git` remote named `origin`. To submit a custom project, set Fetcher to `custom` in `.fossa.yml`")
+	}
+	if uploadBody.Project.Id == "" {
+		return Locator{}, errors.New("Could not infer project name from either `.fossa.yml` or `git` remote named `origin`")
+	}
+	if len(uploadBody.Analysis) == 0 {
+		return Locator{}, errors.New("No data to upload")
+	}
+
+	payload, err := json.Marshal(uploadBody)
+	if err != nil {
+		return Locator{}, errors.Wrap(err, "could not marshal upload data")
+	}
+	log.WithFields(log.Fields{
+		"modules": uploadBody.Analysis,
+		"payload": string(payload),
+	}).Debug("uploading build")
+
+	uploadBody.Meta.v = version.ShortString()
+
+	endpoint, err := url.Parse("/api/cli/v1/build")
+	if err != nil {
+		return Locator{}, errors.New("Failed to generate upload URL")
+	}
+	log.WithField("endpoint", endpoint.String()).Debug("uploading build")
+
+	res, statusCode, err := Post(endpoint.String(), payload)
+	log.WithField("response", res).Debug("build upload completed")
+
+	if statusCode == 428 {
```

Comment:
Is this behaviour still valid in the `v1` API route?

Reply:
Yes, as of now the goal of the v1 API route was to provide an equivalent output for the a different input. The goal was to keep the 428 response.